### PR TITLE
[Sample Code]: Config Tab Authentication- C#

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
Added Authentication in configurable tabs: 
1. Added login AAD Auth
2. log out 
3. Language used: **C#**

### Testing
Configuration Page:
image

**AAD Auth:** 
Config **Page:** for AAD Auth with Sign out button
image

**Silent Auth**
image

### CODE REVIEW

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]
Could you please move this under samples/tab-channel-group-config-page-auth/csharp folder?

**Comment by Trinetra-MSFT** [comment]
> Could you please move this under samples/tab-channel-group-config-page-auth/csharp folder?

Moved this to respective folder

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@
 |5|Channel and group tab quick-start| Sample channel and group tab hello world app.                          |                               |[View][group-tab-quickstart#ts]|[View][group-tab-quickstart#js]
 |6|Channel and group tab with SSO quick-start| Sample channel and group tab with SSO hello world app.        |                               |[View][group-tab-sso-quickstart#ts]|[View][group-tab-sso-quickstart#js]
 |7|SPFx Tab | Sample app showing Microsoft Teams tabs using SharePoint Framework                                    |   | [View][group-channel-tab#ts#spfx] |
-|8|Tab SSO               | Microsoft Teams sample app for tabs Azure AD SSO                                      | | [View][tab-sso#ts] ,[Teams Toolkit](https://docs.microsoft.com/microsoftteams/platform/toolkit/visual-studio-code-tab-sso)
+|8|Tab SSO               | Microsoft Teams sample app for tabs Azure AD SSO  | [View]() | [View][tab-sso#ts] ,[Teams Toolkit](https://docs.microsoft.com/microsoftteams/platform/toolkit/visual-studio-code-tab-sso)
```
Comment (Wajeed-msft) [inline README.md:23]: Could you please add new row for this Config Tab Authentication?

File: samples/tab-channel-group-config-page-auth/csharp/Controllers/HomeController.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            return View();
+        }
+
+        [Route("ConfigureTab")]
+        public IActionResult ConfigureTab()
+        {
+            return View();
+        }
+
+        [Route("AuthStart")]
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/Controllers/HomeController.cs:32]: Could you please separate out Authentication related code to new controller (AuthenticationController) ?

File: samples/tab-channel-group-config-page-auth/csharp/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+   - Create a redirect URI in the format of: https://contoso.ngrok.io/auth/auth-end.
+   - Enable Implicit Grant by selecting Access Tokens and ID Tokens.
+          
+3) Setup a client secret. You will need this when you exchange the token for more API permissions from your backend.
+   - Visit Manage > Certificates & secrets
+   - Create a new client secret.
+          
+4) Setup your API permissions. This is what your application is allowed to request permission to access.
+   - Visit Manage > API Permissions
+   - Make sure you have the following Graph permissions enabled: email, offline_access, openid, profile, and User.Read.
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/README.md:72]: Could you please correct the steps to setup this app? These are steps to configure SSO.

File: samples/tab-channel-group-config-page-auth/csharp/Views/Home/AuthEnd.cshtml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+﻿<!DOCTYPE html>
+<html>
+<head>
+    <title>Simple Authentication Sample Login</title>
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/Views/Home/AuthEnd.cshtml:4]: Could you please add silent auth implementation as well? Provide options to choose from similar to Connector sample.

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/tab-channel-group-config-page-auth/csharp/README.md (hunk 3 above)
Comment (Trinetra-MSFT) [reply samples/tab-channel-group-config-page-auth/csharp/README.md:72]: Resolved

File: samples/tab-channel-group-config-page-auth/csharp/Views/Home/AuthEnd.cshtml (hunk 4 above)
Comment (Trinetra-MSFT) [reply samples/tab-channel-group-config-page-auth/csharp/Views/Home/AuthEnd.cshtml]: Resolved

**Comment by Trinetra-MSFT** [comment]
@Wajeed-msft can we merge this one?

**Comment by Trinetra-MSFT** [comment]
@Wajeed-msft : Included bot AAD and Silent Auth in this sample. Please sign off

File: samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Authentication/SimpleLogOut.cshtml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿
+<!DOCTYPE html>
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Authentication/SimpleLogOut.cshtml:2]: Could you please check the logout implemented here? https://github.com/OfficeDev/microsoft-teams-sample-connector-csharp/blob/master/TeamsToDoAppConnector/Views/Connector/SetupAuth.cshtml#L121

File: samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Home/SetUp.cshtml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        <tr>
+            <td id="greetingMsgSetup">
+               
+            </td>
+        </tr>
+    </table>
+</div>
+
+<script src="@("https://unpkg.com/@microsoft/teams-js@1.3.3/dist/MicrosoftTeams.min.js")"></script>
+<script src="~/Scripts/jquery-1.10.2.js"></script>
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Home/SetUp.cshtml:12]: Could you please remove duplicate scripts tags?

File: samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Home/Index.cshtml [hunk 7]
```diff
@@ -0,0 +1 @@
+﻿<h4>Welcome..</h4>
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Home/Index.cshtml:1]: Please remove if we are not using this file.

File: samples/tab-channel-group-config-page-auth/csharp/README.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+          
+5) Expose an API that will give the Teams desktop, web and mobile clients access to the permissions above
+   - Visit Manage > Expose an API
+   - Add a scope and give it a scope name of access_as_user. Your API url should look like this: api://contoso.ngrok.io/{appID}/access_as_user. In the "who can consent"               step, enable it for "Admins and users". Make sure the state is set to "enabled".
+   - Next, add two client applications. This is for the Teams desktop/mobile clients and the web client.
+      - 5e3ce6c0-2b1f-4285-8d4b-75ee78787346
+      - 1fec8e78-bce4-4aaf-ab1b-5451cc387264
+
+6)  __*This step is specific to Teams.*__
+    - **Edit** the `manifest.json` contained in the  `teamsAppManifest` folder to replace your Microsoft App Id (that was created when you registered your tab earlier) *everywhere* you see the place holder string `<<YOUR-MICROSOFT-APP-ID>>` (depending on the scenario the Microsoft App Id may occur multiple times in the `manifest.json`)
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/README.md:83]: Could you please add Manifest file in the solution?

File: samples/tab-channel-group-config-page-auth/csharp/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+- Visual Studio
+
+- Asp.net Core
+
+## To try this sample
+
+> Note these instructions are for running the sample on your local machine, the tunnelling solution is required because
+
+1) Clone the repository
+    git clone https://github.com/Microsoft/TabAuthentication.git
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/README.md:37]: Please correct the repo path

File: samples/tab-channel-group-config-page-auth/csharp/README.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+2) If you are using Visual Studio
+    - Launch Visual Studio
+    - File -> Open -> Project/Solution
+    - Navigate to `TabAuthentication` folder
```
Comment (Wajeed-msft) [inline samples/tab-channel-group-config-page-auth/csharp/README.md:42]: Please provide full path

File: samples/tab-channel-group-config-page-auth/csharp/README.md (hunk 3 above)
Comment (Wajeed-msft) [reply samples/tab-channel-group-config-page-auth/csharp/README.md:72]: I still see reference of SSO.. 

> Our SSO flow will give you access to the first 4 permissions, and we will have to exchange the token server-side to get an elevated token for the profile permission              (for example, if we want access to the user's profile photo).

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Authentication/SimpleLogOut.cshtml (hunk 5 above)
Comment (Trinetra-MSFT) [reply samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Authentication/SimpleLogOut.cshtml:2]: We have implemented the logout differently

File: samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Home/Index.cshtml (hunk 7 above)
Comment (Trinetra-MSFT) [reply samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Home/Index.cshtml]: Done

File: samples/tab-channel-group-config-page-auth/csharp/README.md (hunk 8 above)
Comment (Trinetra-MSFT) [reply samples/tab-channel-group-config-page-auth/csharp/README.md:84]: Added manifest in the solution

File: samples/tab-channel-group-config-page-auth/csharp/README.md (hunk 10 above)
Comment (Trinetra-MSFT) [reply samples/tab-channel-group-config-page-auth/csharp/README.md]: Updated ReadMe file

**Review by Wajeed-msft — APPROVED** [review]

**Review by asnow003 — APPROVED** [review]

**Pull request merged**